fluss-client/src/main/java/com/alibaba/fluss/client/utils/ConverterUtils.java

```diff
@@ -0,0 +1,610 @@
+                    }
+                    if (value instanceof BigDecimal) {
+                        DecimalType decimalType = (DecimalType) fieldType;
+                        return Decimal.fromBigDecimal(
+                                (BigDecimal) value,
+                                decimalType.getPrecision(),
+                                decimalType.getScale());
+                    } else {
+                        throw new IllegalArgumentException(
+                                String.format(
+                                        "Field %s is not a BigDecimal. Cannot convert to DecimalData.",
+                                        field.getName()));
+                    }
+                };
+            case DATE:
+                return obj -> {
+                    Object value = field.get(obj);
+                    if (value == null) {
+                        return null;
+                    }
+                    if (value instanceof LocalDate) {
+                        return (int) ((LocalDate) value).toEpochDay();
+                    } else {
+                        LOG.warn(
+                                "Field {} is not a LocalDate. Cannot convert to int days.",
+                                field.getName());
+                        return null;
+                    }
+                };
+            case TIME_WITHOUT_TIME_ZONE:
+                return obj -> {
+                    Object value = field.get(obj);
+                    if (value == null) {
+                        return null;
+                    }
+                    if (value instanceof LocalTime) {
+                        LocalTime localTime = (LocalTime) value;
+                        return (int) (localTime.toNanoOfDay() / 1_000_000);
+                    } else {
+                        LOG.warn(
+                                "Field {} is not a LocalTime. Cannot convert to int millis.",
+                                field.getName());
+                        return null;
+                    }
+                };
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                return obj -> {
+                    Object value = field.get(obj);
+                    if (value == null) {
+                        return null;
+                    }
+                    if (value instanceof LocalDateTime) {
+                        return TimestampNtz.fromLocalDateTime((LocalDateTime) value);
+                    } else {
+                        LOG.warn(
+                                "Field {} is not a LocalDateTime. Cannot convert to TimestampData.",
+                                field.getName());
+                        return null;
+                    }
+                };
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                return obj -> {
+                    Object value = field.get(obj);
+                    if (value == null) {
+                        return null;
+                    }
```

Comment:
We should reuse the logic of get field value and null checks, and makes the conversion focus on not_null_object -> not_null_object conversion. You can do this by add a nullalbe wrapper on the top of a not-null wrapper.